libraries/AP_Logger/AP_Logger_File.cpp

```diff
@@ -135,6 +138,17 @@ void AP_Logger_File::periodic_1Hz()
         stop_logging();
     }
 
+    if (_write_fd == -1 && _read_fd == -1 &&
+        logging_enabled() &&
+        !recent_open_error() &&
+        !hal.util->get_soft_armed()) {
+        // retry logging open. This allows for booting with
+        // LOG_DISARMED=1 with a bad microSD or no microSD. Once a
+        // card is inserted then logging starts
+        _initialised = true;
```

Comment:
Forcing this doesn't seem entirely safe at first glance - have we necessarily got a valid _writebuf pointer?

Reply:
I think it is protected by the checks for _write_fd and _read_fd being -1